Rubberduck.Parsing/Rewriter/RewriterInfo/RewriterInfoFinderBase.cs

```diff
@@ -21,36 +21,66 @@ protected static RewriterInfo GetRewriterInfoForTargetRemovedFromListStmt(IToken
             return new RewriterInfo(startIndex, stopIndex);
         }
 
-        protected static int FindStopTokenIndex<TParent>(IReadOnlyList<ParserRuleContext> items, ParserRuleContext item, TParent parent)
+        protected static int FindStopTokenIndexForRemoval(VBAParser.ModuleDeclarationsElementContext element)
         {
-            for (var i = 0; i < items.Count; i++)
+            if (!element.TryGetFollowingContext<VBAParser.IndividualNonEOFEndOfStatementContext>(out var followingIndividualEndOfLineStatement))
             {
-                if (items[i] != item)
-                {
-                    continue;
-                }
-                return FindStopTokenIndex((dynamic)parent, i);
+                return element.Stop.TokenIndex;
             }
 
-            return item.Stop.TokenIndex;
+            //If the endOfStatement starts with a statement separator, it is safe to simply remove that. 
+            if (followingIndividualEndOfLineStatement.COLON() != null)
+            {
+                return followingIndividualEndOfLineStatement.Stop.TokenIndex;
+            }
+
+            //Since there is no statement separator, the individual endOfStatement must contain an endOfLine.
+            var endOfLine = followingIndividualEndOfLineStatement.endOfLine();
+
+            //EndOfLines contain preceding comments. So, we cannot remove the line, if there is one.
+            if (endOfLine.commentOrAnnotation() != null)
+            {
+                return endOfLine.commentOrAnnotation().Start.TokenIndex - 1;
+            }
+
+            return followingIndividualEndOfLineStatement.Stop.TokenIndex;
+        }
+
+        protected static int FindStopTokenIndexForRemoval(VBAParser.MainBlockStmtContext mainBlockStmt)
+        {
+            return FindStopTokenIndexForRemoval((VBAParser.BlockStmtContext)mainBlockStmt.Parent);
         }
 
-        protected static int FindStopTokenIndex(IReadOnlyList<VBAParser.BlockStmtContext> blockStmts, VBAParser.MainBlockStmtContext mainBlockStmt, VBAParser.BlockContext block)
+        protected static int FindStopTokenIndexForRemoval(VBAParser.BlockStmtContext blockStmt)
```

Comment:
maybe note in a comment why we can't unify the two implementations of this method.

Reply:
Will add a comment.